### PR TITLE
Renovation: odata/store es6 refactoring

### PR DESCRIPTION
**Pull request by LazyLahtak**
Part of refactoring odata/* components

(In plans - get rid of odata/mixins)

### CODE REVIEW

File: js/data/odata/store.js [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
+import './query_adapter';
 
 const ANONYMOUS_KEY_NAME = '5d46402c-7899-4ea9-bd81-8b73c47c7683';
 
-function expandKeyType(key, keyType) {
-    const result = {};
-    result[key] = keyType;
-    return result;
-}
+const expandKeyType = (key, keyType) => {
```
Comment (churkin) [inline js/data/odata/store.js]:
```javascript
const expandKeyType = (key, keyType) => { [key]: keyType };
```

File: js/data/odata/store.js [hunk 2]
```diff
@@ -251,21 +242,21 @@ const ODataStore = Store.inherit({
         return d.promise();
     },
 
-    _removeImpl: function(key) {
+    _removeImpl(key) {
         const d = new Deferred();
 
         when(
             this._sendRequest(this._byKeyUrl(key), 'DELETE')
         ).done(
```
Comment (churkin) [inline js/data/odata/store.js]:
```javascript
when(this._sendRequest(this._byKeyUrl(key), 'DELETE'))
  .done(() => d.resolve(key))
  .fail(d.reject);
```

**Review by churkin — CHANGES_REQUESTED** [review]

**Review by MaximKudriavtsev — DISMISSED** [review]

**Comment by LazyLahtak** [comment]
Ready to review

File: js/data/odata/store.js [hunk 3]
```diff
@@ -175,12 +169,9 @@ const ODataStore = Store.inherit({
     * @param1 loadOptions:object
     * @return object
     */
-    createQuery: function(loadOptions) {
+    createQuery(loadOptions = {}) {
```
Comment (IlyaKhD) [inline js/data/odata/store.js]:
```suggestion
    createQuery(loadOptions) {
```

File: js/data/odata/store.js [hunk 4]
```diff
@@ -195,21 +186,17 @@ const ODataStore = Store.inherit({
         };
 
         // NOTE: For AppBuilder, do not remove
-        if(isDefined(loadOptions.urlOverride)) {
-            url = loadOptions.urlOverride;
-        } else {
-            url = this._url;
-        }
+        url = loadOptions.urlOverride ?? this._url;
```
Comment (IlyaKhD) [inline js/data/odata/store.js]:
```suggestion
        url = loadOptions && loadOptions.urlOverride ?? this._url;
```

Or if we enabled optional chaining as well as coalescing:
```suggestion
        url = loadOptions?.urlOverride ?? this._url;
```

Comment (IlyaKhD) [reply js/data/odata/store.js]: Same for ` if(loadOptions.customQueryParams) {` below

File: js/data/odata/store.js [hunk 5]
```diff
@@ -218,54 +205,48 @@ const ODataStore = Store.inherit({
...
-        const that = this;
         const d = new Deferred();
 
         when(this._sendRequest(this._url, 'POST', null, values))
-            .done(function(serverResponse) {
-                d.resolve(config().useLegacyStoreResult ? values : (serverResponse || values), that.keyOf(serverResponse));
-            })
+            .done(serverResponse =>
+                d.resolve(
+                    config().useLegacyStoreResult ? values : serverResponse || values,
```
Comment (IlyaKhD) [inline js/data/odata/store.js]:
```suggestion
                    serverResponse && !config().useLegacyStoreResult ? serverResponse : values,
```

File: js/data/odata/store.js [hunk 6]
```diff
@@ -218,54 +205,48 @@ const ODataStore = Store.inherit({
...
+                    : d.resolve(serverResponse || values, key)
+            )
+            .fail(d.reject);
 
         return d.promise();
     },
 
-    _removeImpl: function(key) {
+    _removeImpl(key) {
         const d = new Deferred();
```
Comment (churkin) [inline js/data/odata/store.js:163]:
```javascript
return this._sendRequest(this._byKeyUrl(key), 'DELETE');
```
?

**Review by churkin — CHANGES_REQUESTED** [review]

File: js/data/odata/store.js (hunk 6 above)
Comment (LazyLahtak) [reply js/data/odata/store.js:163]: We can't get rid of this construction. The `_sendRequest` return `Deffered` object but on success we return
```js
() => d.resolve(key)
```
This `resolve` depends on the `key` value, but `_sendRequest` the `Deffered` object recieves from `_sendRequest` - not

**Comment by LazyLahtak** [comment]
Ready to review

**Review by MaximKudriavtsev — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**